kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -260,4 +286,98 @@ private static Predicate reverseComparatorFilter(Predicate predicate) {
+            // F N F T T
+            // N N N N N
+            case "AND":
+                return constructDataSkippingFilter(
+                    new Or(
+                        new Predicate("NOT", asPredicate(getLeft(childPredicate))),
+                        new Predicate("NOT", asPredicate(getRight(childPredicate)))
+                    ),
+                    schemaHelper
+                );
+
+            // Similar to AND, we can (and want to) push the NOT past the OR using deMorgan's law.
+            case "OR":
+                return constructDataSkippingFilter(
+                    new And(
+                        new Predicate("NOT", asPredicate(getLeft(childPredicate))),
+                        new Predicate("NOT", asPredicate(getRight(childPredicate)))
+                    ),
+                    schemaHelper
+                );
+
+            case "=":
+                Expression left = getLeft(childPredicate);
+                Expression right = getRight(childPredicate);
+                if (left instanceof Column && right instanceof Literal) {
+                    Column leftCol = (Column) left;
+                    Literal rightLit = (Literal) right;
+                    if (schemaHelper.isSkippingEligibleMinMaxColumn(leftCol) &&
+                        schemaHelper.isSkippingEligibleLiteral(rightLit)) {
+                        // Match any file whose min/max range contains anything other than the
+                        // rejected point.
+                        // For example a = 1 --> minValue.a <= 1 AND 1 <= maxValue.a
+                        return Optional.of(
+                            new Or(
+                                new Predicate("<", schemaHelper.getMinColumn(leftCol), rightLit),
+                                new Predicate("<", rightLit, schemaHelper.getMaxColumn(leftCol))
+                            )
+                        );
+                    }
+                } else if (right instanceof Column && left instanceof Literal) {
+                    return constructDataSkippingFilter(
+                        new Predicate("NOT", new Predicate("=", right, left)),
+                        schemaHelper);
+                }
+                break;
+            case "<":
+                return constructDataSkippingFilter(
+                    new Predicate(">=", childPredicate.getChildren()),
+                    schemaHelper);
+            case "<=":
+                return constructDataSkippingFilter(
+                    new Predicate(">", childPredicate.getChildren()),
+                    schemaHelper);
+            case ">":
+                return constructDataSkippingFilter(
+                    new Predicate("<=", childPredicate.getChildren()),
+                    schemaHelper);
+            case ">=":
+                return constructDataSkippingFilter(
+                    new Predicate("<", childPredicate.getChildren()),
+                    schemaHelper);
```

Comment:
can we use the `reverseComparatorFilter` here?

Reply:
This diff is from the previous PR actually, but also we can't (although we cold make something similar). reverseComparatorFilter just reverses the left & right argument; here we are taking the not.
So for reverse we have 1 > a --> a < 1
But for the not we have 1 > a --> 1 <= a